### PR TITLE
Add the device type trait specialization for Events

### LINKED ISSUE #1428: Add DevType<Event> trait
Hi,
for simplifying some generic code we are writing, we would benefit for traits to extract the `Device` type from the `Event` type, something that returns `TDev` for `EventGenericThreads<TDev>` and `DevUniformCudaHipRt` for `EventUniformCudaHip`.

Of course we can add such  trait in our code, but it could come in useful for other projects as well.

### PR DESCRIPTION
**Pull request by fwyzard**

### CODE REVIEW

**Comment by fwyzard** [comment]
Fixes #1428 .

File: include/alpaka/event/EventUniformCudaHipRt.hpp [hunk 1]
```diff
@@ -123,6 +123,12 @@ namespace alpaka
     };
     namespace traits
     {
+        //! The CUDA/HIP RT device event device type trait specialization.
+        template <>
```
Comment (BenjaminW3) [inline include/alpaka/event/EventUniformCudaHipRt.hpp]: clang-format says:
```suggestion
        template<>
```

**Review by BenjaminW3 — DISMISSED** [review]

File: include/alpaka/event/EventUniformCudaHipRt.hpp (hunk 1 above)
Comment (fwyzard) [reply include/alpaka/event/EventUniformCudaHipRt.hpp]: what version of clang do you use ?
`clang-format` from clang 11 complains about unknown options in the format file :-(

Comment (bernhardmgruber) [reply include/alpaka/event/EventUniformCudaHipRt.hpp]: We use clang-format 12 (specifically 12.0.1), as documented [here](https://github.com/alpaka-group/alpaka/blob/develop/CONTRIBUTING.md).

Comment (fwyzard) [reply include/alpaka/event/EventUniformCudaHipRt.hpp]: Fixed, thanks.

**Review by bernhardmgruber — APPROVED** [review]

**Pull request merged**